### ISSUE TITLE
Support iPhone X

### ISSUE THREAD

**Issue by TimOliver** [TimOliver/TOCropViewController#196]
Jumping in here before anyone else does.

Due to the extremely custom layout of this view controller, it is hilariously broken on iPhone X.

![simulator screen shot - iphone x - 2017-09-13 at 15 14 53](https://user-images.githubusercontent.com/429119/30360758-8f12640a-9896-11e7-8693-b409fe2fc11a.png)

For the most part, it should just be a matter of making the tool bar and the crop box insets obey the safe area of iOS 11. It shouldn't be TOO bad to fix.

**Comment by mugunth87** [comment]
@TimOliver When will this issue be fixed?

**Comment by TimOliver** [comment]
It's virtually finished in `master` already. The only thing remaining is I need to set up the toolbar background to extend under the notch, but even that isn't too much of a problem since it isn't noticable.

I plan to close this issue off after I've finished that part. But I've got some other pressing stuff, so I won't commit to any timeframe except 'before the iPhone X is released'. :)

**Comment by inPhilly** [comment]
The iPhone X is due out November 3. I am working on another issue with this repository, but I was also hoping to support iPhone X in my app that will be released in the next couple of weeks. Do you happen to have any updates on this? Thanks.

**Comment by TimOliver** [comment]
Nope. No updates at the moment sorry. I'm at multiple conferences this week
without my dev setup. XD

**Comment by TimOliver** [comment]
Okay. I'm heading home from my conference week now. I had a quick chance to fill in the rest of the iPhone X logic I was missing.

It's all checked into `master` now. Try pulling it down and see how it looks to you!

**Comment by Theome** [comment]
The notch or the system bar overlaps the UI in some situations. This issue affects all devices (iPhone X, iPad, iPhone 8).
<img width="819" alt="bildschirmfoto 2017-11-01 um 12 46 15" src="https://user-images.githubusercontent.com/668256/32273537-1a551f3e-bf03-11e7-8145-91d7b24946a8.png">
<img width="421" alt="bildschirmfoto 2017-11-01 um 12 46 35" src="https://user-images.githubusercontent.com/668256/32273539-1a6ff728-bf03-11e7-8e67-aecad1a4222f.png">
<img width="419" alt="bildschirmfoto 2017-11-01 um 12 49 03" src="https://user-images.githubusercontent.com/668256/32273540-1a8b4a82-bf03-11e7-9cb8-79818e26923d.png">

**Comment by yojique** [comment]
sorry for bothering. any good news on the issue?

**Comment by inPhilly** [comment]
Sorry to bother again Tim. Any updates with this issue? Thanks for this great controller, again.

**Comment by TimOliver** [comment]
Sorry for the delay folks! I was Mac-less for all of last week, but I'm back in business now!

Oh jeez, okay. I see the issue. The top inset isn't account for the notch properly still on iPhone X, and not even handling the status bar properly on other iOS devices.

I think I know how to fix it. I'll see if I can smooth it all out today. :)

**Comment by Theome** [comment]
Hi Tim, thanks for the update! With the new version from `master` (commit `4835a0`), it looks correct on portrait under iOS 11. Configurations that use landscape or iOS 10 still look wrong.

**Comment by TimOliver** [comment]
Alright. It's done. I ended up needing to spend a whole day on it, but it's finally done. :D

![img_4507](https://user-images.githubusercontent.com/429119/32903327-19d27b2e-cb38-11e7-94eb-cec301292780.JPG)

There were a lot of little edge cases that broke by sheer virtue of the fact that the iPhone X notch introduces a pretty substantial offset. A lot of images broke when performing rotation animations on them and the likes.

I tested it very thoroughly on my iPhone X, my iPad and iOS 10.3 in the Simulator, and I'm happy with it.

It's been pushed to CocoaPods under version 2.2. Enjoy!
